src/aatoolbox/datasources/iri/iri_seasonal_forecast.py

```diff
@@ -0,0 +1,269 @@
+        # its data on, which is 1 degree resolution
+        # non-rounded coordinates can be given to the URL which then
+        # automatically rounds them, but for file saving we prefer to do
+        # this ourselves
+        geo_bounding_box.round_coords(round_val=1, offset_val=0)
+        self._geobb = geo_bounding_box
+        self._forecast_type = forecast_type
+
+    def download(
+        self,
+        iri_auth: str,
+        clobber: bool = False,
+    ):
+        """
+        Download the IRI seasonal tercile forecast as NetCDF file.
+
+        Parameters
+        ----------
+        iri_auth: str
+            iri key for authentication. An account is
+            needed to get this key config. For an account this key might
+            be changed over time, so might need to update it regularly
+        clobber : bool, default = False
+            If True, overwrites existing raw files
+        """
+        output_filepath = self._get_raw_path()
+        output_filepath.parent.mkdir(parents=True, exist_ok=True)
+
+        url = self._get_url()
+        logger.info("Downloading IRI NetCDF file.")
+        return self._download(
+            filepath=output_filepath,
+            url=url,
+            iri_auth=iri_auth,
+            clobber=clobber,
+        )
+
+    @staticmethod
+    @check_file_existence
+    def _download(filepath, url, iri_auth, clobber):
+        response = requests.get(
+            url,
+            # have to authenticate by using a cookie
+            cookies={"__dlauth_id": iri_auth},
+        )
+        with open(filepath, "wb") as out_file:
+            out_file.write(response.content)
+        return filepath
+
+    def process(self, clobber: bool = False):
+        """
+        Process the IRI forecast.
+
+        Parameters
+        ----------
+        clobber : bool, default = False
+            If True, overwrites existing processed files
+
+        """
+        ds = xr.load_dataset(
+            self._get_raw_path(),
+            decode_times=False,
+            drop_variables="C",
+        )
+        processed_file_path = self._get_processed_path()
+        processed_file_path.parent.mkdir(parents=True, exist_ok=True)
+        return self._process(
+            filepath=processed_file_path, ds=ds, clobber=clobber
+        )
+
+    @staticmethod
+    @check_file_existence
+    def _process(filepath: str, ds, clobber: bool):
+        # fix dates
+        ds.aat.set_time_dim(t_dim="F", inplace=True)
+        ds.aat.correct_calendar(inplace=True)
+        ds = xr.decode_cf(ds)
+
+        # IRI downloads in the order you give the coordinates
+        # so make sure to invert them
+        # IRI accepts -180 to 180 longitudes and 0 to 360
+        # but automatically converts them to -180 to 180
+        # so we don't need to do that
+        # TODO: can be removed once we have a check in the
+        # geoboundingbox class for south<north
+        # TODO: for some reason the `inplace` is not working
```

Comment:
Hm maybe @caldwellst can look into this?

Reply:
Yes discussed and made an issue on it